htdocs/core/tpl/objectline_view.tpl.php

```diff
@@ -166,7 +166,7 @@
 	}
 }
 
-if ($user->rights->fournisseur->lire && $line->fk_fournprice > 0)
+if ($user->rights->fournisseur->lire && $line->fk_fournprice > 0 && !empty($conf->global->DISPLAY_SUPPLIER_OBJECTLINES))
```

Comment:
Why not. However, because default behaviour is to show, the hidden option must hide.
So    
`&& empty($conf->global->SUPPLIER_HIDE_SUPPLIER_OBJECTLINES))`